pandas/core/sorting.py

```diff
@@ -241,7 +241,23 @@ def nargsort(items, kind='quicksort', ascending=True, na_position='last'):
 
     # specially handle Categorical
     if is_categorical_dtype(items):
-        return items.argsort(ascending=ascending, kind=kind)
+        mask = isna(items)
+        null_idx = np.where(mask)[0]
+        sorted_idx = items.argsort(ascending=ascending, kind=kind)
```

Comment:
do we not have a .sort_values on Categorical that accepts na_position already?

if we don’t all of this code should live there anyhow

Reply:
the way I see it, `df.sort_values` is not calling `sort_values` method from column objects, it's calling `nargsort`. Since `pd.Series` also doesn't have `nargsort`, I feel it may fit better in here rather than `Categorical.nargsort`